Index: src/main/java/com/example/demo/service/UserService.java
===================================================================
diff --git a/src/main/java/com/example/demo/service/UserService.java b/src/main/java/com/example/demo/service/UserService.java
--- a/src/main/java/com/example/demo/service/UserService.java	
+++ b/src/main/java/com/example/demo/service/UserService.java	
@@ -19,18 +19,18 @@
 
 @Service
 @RequiredArgsConstructor
+@Transactional
 public class UserService {
 
+    @Autowired
     private final UserRepositoryInterface userRepository;
+
     @Autowired
     private final RoleRepositoryInterface roleRepository;
 
     public UpdateResponse update(String username, UpdateRequest request) throws UsernameNotFoundException {
         var user = userRepository.findByUsername(username).orElseThrow(()
                 -> new UsernameNotFoundException("User with username " + username + " not found"));
-        for(Role role:user.getRoles()) {
-            removeRoleFromUser(user, role);
-        }
         user.setMobileNumber(request.getMobile());
         user.setEmail(request.getEmail());
         user.setRoles(generateRoles(request.getRoles()));
@@ -41,14 +41,6 @@
                 .build();
     }
 
-    @Transactional
-    public void removeRoleFromUser(User user, Role role) {
-        Set<Role> roles = user.getRoles();
-        roles.remove(role);
-        user.setRoles(roles);
-        userRepository.save(user);
-    }
-
     public Set<Role> generateRoles(List<String> rolesData) {
         Set<Role> rolesSet = new HashSet<>();
         for(String role: rolesData) {
